src/payload/collections/Elections/endpoints/results.ts

```diff
@@ -0,0 +1,57 @@
+import type { PayloadHandler } from 'payload/config'
+
+import { checkRole } from '../../Users/checkRole'
+
+export const electionResults: PayloadHandler = async (req, res): Promise<void> => {
+  const { user, payload } = req
+
+  if (!user || !checkRole(['admin'], user)) {
+    res.status(401).json({ error: 'Unauthorized' })
+    return
+  }
+
+  const electionResult = await payload.find({
+    collection: 'electionResults',
+    depth: 2,
+    where: {
+      election: {
+        equals: req.params.id,
+      },
+    },
+    pagination: false,
+  })
+
```

Comment:
You can add `user` to payload find.
Since we are directly returning the docs we shouldn't return stuff the user shouldn't be able to see (even if they are admin)